src/Agent.Plugins/PipelineArtifact/DedupManifestArtifactClientFactory.cs

```diff
@@ -22,14 +23,26 @@ public static class DedupManifestArtifactClientFactory
     {
         private static readonly int DedupStoreClientMaxParallelism = 16 * Environment.ProcessorCount;
 
+        // Use with servers that don't require custom telemetry.
         public static DedupManifestArtifactClient CreateDedupManifestClient(AgentTaskPluginExecutionContext context, VssConnection connection)
         {
             var dedupStoreHttpClient = connection.GetClient<DedupStoreHttpClient>();
             var tracer = new CallbackAppTraceSource(str => context.Output(str), SourceLevels.Information);
             dedupStoreHttpClient.SetTracer(tracer);
             var client = new DedupStoreClientWithDataport(dedupStoreHttpClient, DedupStoreClientMaxParallelism);
-            var dedupManifestClient = new DedupManifestArtifactClient(client, tracer);
-            return dedupManifestClient;
+            var telemetry = new BlobStoreClientTelemetry(tracer, dedupStoreHttpClient.BaseAddress);
+            return new DedupManifestArtifactClient(telemetry, client, tracer);
+        }
```

Comment:
we can just use one method as we don't support this as a client library to be exported to users